lib/ret/hub.ex

```diff
@@ -279,14 +288,60 @@ defmodule Ret.Hub do
     end
   end
 
+  defp add_default_perms_to_changeset(changeset) do
+    default_perms = %{
+      spawn_and_move_media: true,
+      spawn_camera: true,
+      spawn_drawing: true,
+      pin_objects: true
+    }
+
+    changeset |> put_change(:permissions, default_perms |> hub_perms_to_int!)
+  end
+
+  @hub_perms %{
+    0x0000_0001 => :spawn_and_move_media,
+    0x0000_0002 => :spawn_camera,
+    0x0000_0004 => :spawn_drawing,
+    0x0000_0008 => :pin_objects
+  }
+
+  @hub_perms_keys @hub_perms |> Map.values()
+
+  def hub_perms_to_int!(%{} = perms) do
+    invalid_perms = perms |> Map.drop(@hub_perms_keys) |> Map.keys()
+
+    if invalid_perms |> Enum.count() > 0 do
+      raise ArgumentError, "Invalid permissions #{invalid_perms |> Enum.join(", ")}"
+    end
+
+    @hub_perms |> Enum.reduce(0, fn {val, perm}, acc -> if(perms[perm], do: 1, else: 0) * val + acc end)
+  end
+
+  def has_perm!(%Hub{permissions: hub_perms_bit_field}, perm) do
```

Comment:
i find this function name fairly confusing, since the `!` implies to me it will throw if the permission is not granted -- I'd name it `has_perm?`. i think bang is reserved for error conditions, not argument validation (where, for example, you'd typically match on err for the non-bang version)

also if we decide to rename the column, I'd call this `has_member_perm?` or something aligned with the new column name